src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.hide();
```

Comment:
The `bite` method should decrease the health of the Herbivore by 50 if it's not hidden and is alive. After decreasing the health, it should check if the animal's health is less than or equal to 0 and remove it from the `Animal.alive` array if necessary. Currently, the method is only toggling the `hidden` state which is not the expected behavior.